esphome/components/ntcsim/ntcsim.cpp

```diff
@@ -0,0 +1,48 @@
+#include "ntcsim.h"
+
+#include "esphome/core/log.h"
+
+namespace esphome {
+namespace ntcsim {
+
+static const char *const TAG = "ntcsimulation";
+
+void NTCSimulation::setup() {
+  sensor_->add_on_state_callback([this](float value) { process_(value); });
+  if (sensor_->has_state())
+    process_(sensor_->state);
+}
+void NTCSimulation::dump_config() {
+  LOG_SENSOR("", "NTC Simulation", this);
+  ESP_LOGI(TAG, "  Coefficient a: %.8f", a_);
+  ESP_LOGI(TAG, "  Coefficient b: %.8f", b_);
+  ESP_LOGI(TAG, "  Coefficient c: %.8f", c_);
+}
+
+float NTCSimulation::get_setup_priority() const { return setup_priority::DATA; }
+
+void NTCSimulation::process_(float value) {
+  if (std::isnan(value)) {
+    publish_state(NAN);
+    return;
+  }
```

Comment:
```suggestion
  if (!std::isfinite(value)) {
    this->publish_state(NAN);
    return;
  }
```

Use this if you don't accept -INF / +INF values and NAN.